src/PHPHtmlParser/Dom.php

```diff
@@ -525,34 +532,34 @@ protected function clean(string $str): string
         $str = str_replace(["\r\n", "\r", "\n"], $replace, $str);
 
         // strip the doctype
-        $str = mb_eregi_replace("<!doctype(.*?)>", '', $str);
+        $str = (string) mb_eregi_replace("<!doctype(.*?)>", '', $str);
 
         // strip out comments
-        $str = mb_eregi_replace("<!--(.*?)-->", '', $str);
+        $str = (string) mb_eregi_replace("<!--(.*?)-->", '', $str);
```

Comment:
mb_eregi_* will return a non-string in the case of errors. I do not want to convert the false to an empty string and continue, that would be some very strange behavior for the developers. I would remove all the casts to the mb_eregi methods.

Reply:
Well, the idea of this change is that if one mb_eregi_replace function return false, next mb_eregi_replace in place of $str will be false, not string, which this function expect. This can couse of emmitint some errors/warnings, but result of next mb_eregi_replace (that with boolean instead of string) will result same.

Reply:
I would rather an error be thrown then they get an empty string as part of the result.

Reply:
But it would be BC change. For now adding type casting do not change behavior of script. For now clean method return string type cast, so even if some mb_eregi_replace will return false (which is very unlikely), function will return empty string. I do not insist to put type cast on mb_eregi_replace calls, but I think while it do not change method behavior at all, it's more clear to know how this function behave.